### DISCUSSION TITLE
What Am I Doing Wrong Now?

### DISCUSSION THREAD

**Discussion by Donostio** [cecobask/imdb-trakt-sync#58] [General]
##[debug]Starting: Sync
##[debug]Loading inputs
##[debug]Loading env
Run make sync
##[debug]/usr/bin/bash -e /home/runner/work/_temp/2be9a39f-ea9a-46e6-8a5f-c[7](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10165549459/job/28113915012#step:6:7)a172689da4.sh
{"time":"2024-07-30T15:52:55.39543[8](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10165549459/job/28113915012#step:6:8)472Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":64},"msg":"launched new browser instance","url":"ws://127.0.0.1:37815/devtools/browser/d56f0db1-6d05-4e07-b16c-55d3eb3378d3","headless":true,"trace":false}
{"time":"2024-07-30T15:53:13.06631772[9](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10165549459/job/28113915012#step:6:9)Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).hydrate","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":197},"msg":"hydrated imdb client","username":"shearin-gary","userID":"ur[31](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10165549459/job/28113915012#step:6:32)184890","watchlistID":"ls008299431","lists":[" ***\n"]}
Error: error creating syncer: failure initialising trakt client: failure hydrating client: failure generating auth codes: http request POST https://api.trakt.tv/oauth/device/code returned status code 500: unexpected status code 500
make: *** [Makefile:10: sync] Error 1
Error: Process completed with exit code 2.
##[debug]Finishing: Sync

**Comment by cecobask** [comment]
Something looks fishy in your configuration. The logs reported that your list ids are set to an invalid value. Please share what’s the exact value you’ve set for the `IMDB_LISTS` secret. Its value should be comma-separated list of IDs.

**Comment by Donostio** [comment]
Thank you. 
I had it in CSV format before, but switched it (to try and get things working) to an array with space - space in front of each on new lines. I have now switched back to CSV:
ls042296906,ls095507165,ls042450046,ls095507882,ls042408142,ls095507498,ls529418213

But I'm getting this:
##[debug]Evaluating condition for step: 'Sync'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Sync
##[debug]Loading inputs
##[debug]Loading env
Run make sync
##[debug]/usr/bin/bash -e /home/runner/work/_temp/0b65a[2](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:2)28-da23-4665-a1ef-2ebc26067bb5.sh
{"time":"2024-07-[3](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:3)1T08:03:42.175009895Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":6[4](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:4)},"msg":"launched new browser instance","url":"ws://127.0.0.1:46783/devtools/browser/78776b0[5](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:5)-0fed-40a5-ad61-888b87a2[6](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:6)399","headless":true,"trace":false}
{"time":"2024-0[7](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:7)-31T08:06:05.74757682Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).hydrate","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":197},"msg":"hydrated imdb client","username":"shearin-gary","userID":"ur311[8](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:8)4890","watchlistID":"ls0082[9](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:9)9431","lists":["ls042296906","ls095507165","ls042450046","ls095507882","ls042408142","ls095507498","ls529418213\n"]}
{"time":"2024-07-[31](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:32)T08:06:06.980759095Z","level":"WARN","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*TraktClient).doRequest","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/trakt.go","line":310},"msg":"trakt rate limit reached, waiting for 3s then retrying http request POST https://trakt.tv/activate/authorize"}
{"time":"2024-07-31T08:06:10.207780118Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/internal/syncer.(*Syncer).Sync","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/internal/syncer/syncer.go","line":62},"msg":"sync started"}
{"time":"2024-07-31T08:06:18.449065549Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls095507882"}
{"time":"2024-07-31T08:06:20.929421816Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls042450046"}
{"time":"2024-07-31T08:06:21.224316615Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls095507498"}
{"time":"2024-07-31T08:06:21.404593927Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls042296906"}
{"time":"2024-07-31T08:06:21.711561668Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls095507165"}
{"time":"2024-07-31T08:06:22.249007213Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls529418213\n"}
{"time":"2024-07-31T08:06:22.3181609[35](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10171604343/job/28145671850#step:6:36)Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":221},"msg":"exported list","id":"ls042408142"}
Error: error performing sync: failure fetching imdb lists: failure fetching exported resources: failure waiting for exports to become available: failure filtering elements by js selector: eval js error: SyntaxError: Invalid or unexpected token <nil>
{"time":"2024-07-31T08:06:31.59282798Z","level":"ERROR","source":{"function":"github.com/cecobask/imdb-trakt-sync/internal/syncer.(*Syncer).Sync","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/internal/syncer/syncer.go","line":64},"msg":"failure hydrating imdb client","error":"failure fetching imdb lists: failure fetching exported resources: failure waiting for exports to become available: failure filtering elements by js selector: eval js error: SyntaxError: Invalid or unexpected token <nil>"}
make: *** [Makefile:10: sync] Error 1
Error: Process completed with exit code 2.
##[debug]Finishing: Sync

**Comment by cecobask** [comment]
The last item in your lists secret is invalid - `ls529418213\n`. Just remove the `\n` part and you should be good.

**Comment by Donostio** [comment]
Thank you! Finally sorted. I think the \n must have been some accidentally copied formatting

**Comment by cecobask** [comment]
Added a safeguard for this scenario in the latest release: https://github.com/cecobask/imdb-trakt-sync/releases/tag/v1.1.1

**Comment by Donostio** [comment]
I updated the fork yesterday. Do I also need to change some settings / secrets?  
Thank you.
##[debug]Evaluating condition for step: 'Sync'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Sync
##[debug]Loading inputs
##[debug]Loading env
Run make sync
##[debug]/usr/bin/bash -e /home/runner/work/_temp/c55[2](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:2)954a-af01-4175-8408-37b48453c88f.sh
{"time":"2024-08-22T15:50:20.258256495Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":64},"msg":"launched new browser instance","url":"ws://127.0.0.1:[3](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:3)5145/devtools/browser/503a2d0c-a927-[4](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:4)eb5-95ce-8d799fe53911","headless":true,"trace":false}
{"time":"2024-08-22T1[5](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:5):50:35.407645728Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).hydrate","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":197},"msg":"hydrated imdb client","username":"shearin-gary","userID":"ur31184890","watchlistID":"ls008299431","lists":["ls04229[6](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:6)906","ls095507165","ls042450046","ls09550[7](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:7)882","ls042408142","ls09550749[8](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:8)","ls529418213"]}
{"time":"2024-08-22T15:50:36.770488[9](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:9)91Z","level":"WARN","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.(*TraktClient).doRequest","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/trakt.go","line":[30](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:31)7},"msg":"trakt rate limit reached, waiting for 3s then retrying http request POST https://trakt.tv/activate/authorize"}
{"time":"2024-08-22T15:50:40.01747529Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/internal/syncer.(*Syncer).Sync","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/internal/syncer/syncer.go","line":62},"msg":"sync started"}
panic: context deadline exceeded

goroutine 103 [running]:
github.com/go-rod/rod/lib/utils.init.func2({0xaaeb40?, 0x13ddf20?})
	/home/runner/go/pkg/mod/github.com/go-rod/rod@v0.116.1/lib/utils/utils.go:68 +0x1d
github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient.New.(*Browser).WithPanic.genE.func1({0xc00066ea60?, 0xc0003ae550?, 0xc0002d4c30?})
	/home/runner/go/pkg/mod/github.com/go-rod/rod@v0.116.1/must.go:36 +0x62
github.com/go-rod/rod.(*Page).MustClose(0xc0002daf20)
	/home/runner/go/pkg/mod/github.com/go-rod/rod@v0.116.1/must.go:[33](https://github.com/Donostio/imdb-trakt-sync/actions/runs/10507686353/job/29122336182#step:6:34)6 +0x7b
github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).navigateAndValidateResponse.func2()
	/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go:528 +0x17
github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).exportResource(0xb5486e?, {0xc0000d8030, 0x25})
	/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go:373 +0x4bc
github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListExport(0xc000284288, {0xc00003015f, 0xb})
	/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go:218 +0xaa
github.com/cecobask/imdb-trakt-sync/pkg/client.(*IMDbClient).ListsExport.func1()
	/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go:229 +0x1f
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:75 +0x96
make: *** [Makefile:10: sync] Error 2
Error: Process completed with exit code 2.

**Comment by cecobask** [comment]
Hi @Donostio, it looks like you're not missing any secrets. Please set the `IMDB_TRACE` repository secret to `true`, so I can get more details.

**Comment by Donostio** [comment]
Thank you, done.
